File: Python/boj/Silver/11047.py
import sys

if __name__ == "__main__":
    try:
        sys.stdin = open("input.txt", "r")
    except FileNotFoundError:
        pass

    N, K = map(int, input().split(" "))
    coins = [int(input()) for _ in range(N)]
    count = 0
    index = N - 1

    while K > 0:
        if K >= coins[index]:
            count = count + K // coins[index]
            K = K % coins[index]
        else:
            index = index - 1

    print(count)

File: JavaScript/boj/Silver/11047.js
const fs = require('fs');
const filePath = process.platform === 'linux' ? '/dev/stdin' : './input.txt';
const input = fs.readFileSync(filePath).toString().trim().split('\n');

let [N, K] = input[0].split(' ').map(Number);
const coins = input.slice(1).map(Number);
let result = 0;

for (let i = N - 1; i >= 0; i--) {
  if (coins[i] > K) continue;

  result += Math.floor(K / coins[i]);
  K %= coins[i];
}

console.log(result);
